### DISCUSSION TITLE
AI Enabled Long Range Messages

### DISCUSSION THREAD

**Discussion by alexanderson1993** [Thorium-Sim/thorium-nova#436] [Ideas]
Just a quick note based on this thing that I just found: https://www.emailtriager.com

tl;dr: You can automatically generate draft emails based on two AI prompts: The contents of the message you are replying to, and what you want the reply to include. 

We could apply this to long range messages. As part of this mission timeline, you can add and remove destination/prompt/response pairs and feed it into an AI to produce the response. 

I want to make it clear, this won't be required. In fact, it will probably be an optional (paid?) add-on. But the goal of Thorium Nova is to provide tools to storytellers. This is a great example of that.

As an example, taken from Intolerance. This is from ChatGPT. A more sophisticated prompt could make it more generic to other missions but the form will mostly be the same:

- Set up the context for the simulation.
- Describe the crew's ship and a brief bit of information about their mission.
- Describe the style of response that you want.
- Not shown here, but we could also include all of the previous messages from the crew so the AI has all of the context.

Okay, here's the prompt:
```
Write me a response to an email written in the 24th century. The email is sent from the USS Odyssey, an exploratory spaceship. Their mission is to meet an escaped slave named Rool and decide whether or not to give Rool political asylum. Rool came through a wormhole.

Your email should meet a 5th grade reading level. Use short, terse sentences without a lot of extra words or language. Your response should be 2 paragraphs maximum.

Use the following format for your email response:

TO: USS Odyssey
FROM: Starbase 74

[message]

Starbase 74 out

The crew's email is as follows:

TO: Starbase 74
FROM: USS Odyssey

Another ship just came through the wormhole. He says he is a "keeper" and is trying to get Rool back. What should we do?

USS Odyssey out
```

And the response:

```
TO: USS Odyssey
FROM: Starbase 74

Greetings, USS Odyssey. Thank you for your email. It is important that you keep Rool safe and do not let the "keeper" take him. It is our duty to provide political asylum to those who need it, and Rool's escape from slavery is a clear indication that he needs our help.

We suggest that you do your best to protect Rool and keep him out of the "keeper's" reach. If necessary, you may need to defend yourselves against the other ship. However, try to avoid violence if possible.

Starbase 74 out.
```

The prompt could be further adjusted based on different situations during the mission, with prompts being added and removed over the course of the mission.

Thoughts?
